### ISSUE TITLE
Argument of type 'any[]' is not assignable to parameter of type '[string]'

### ISSUE THREAD

**Issue by namnv04** [lukeautry/tsoa#322]
I started a brand new tsos project by install latest version of tsoa@2.3.3, write controllers. I am able to gen routes and swagger but when I run `npm run tsc` to compile Typescript to Javascript it gives me error related to the autogenerated routes file:

```
 Argument of type 'any[]' is not assignable to parameter of type '[string]'.
  Property '0' is missing in type 'any[]' but required in type '[string]'.

26             const promise = controller.greetUser.apply(controller, validatedArgs);
                                                                      ~~~~~~~~~~~~~
```

This didn't happen to me when I was using tsoa@2.2.3 

Details error in the screenshot.

Any one has had this issue? and any solution? 

<img width="925" alt="screen shot 2018-12-13 at 10 57 06 am" src="https://user-images.githubusercontent.com/18271103/49950473-e2177280-fec5-11e8-85a6-810ece65d547.png">

**Comment by lukeautry** [comment]
Hmm. What version of TypeScript are you using?

If you look at the error, it seems to be thinking that validatedArgs should be of type [string] (i.e. an array with a single string as the first and only element). I _believe_ this is one of the features of the latest TypeScript release - typed call and apply parameters.

**Comment by namnv04** [comment]
Thanks for quick response. I'm using typescript@3.2.2. I am not using `strictBindCallApply`. I just tried using and it doesn't work either. 

I have and old project that I started a few months ago with old `package-lock.json` and I am able to upgrade the tsoa version to latest. But if I setup tsoa from scratch, it doesn't work.

Since the routes.ts file are auto generated, I can't make change to it. Can you think of any workaround?

**Comment by lukeautry** [comment]
How sure are you that a new routes file is being generated? i.e. if you delete routes.ts, and regenerate it, does it still generate the same bad code?

Is this project open source so I can look at it?

**Comment by namnv04** [comment]
The route generating is working I did update controller file and new route file get generated with new routes. 

Here's the code base that I setup for test: https://github.com/namnv04/tsoa-sample

Few command to run: 

```
npm install
npm run gen-routes
npm run tsc (this you will see the error)
```

Thanks!

**Comment by Dancia** [comment]
Same problem here.

**Comment by briepace** [comment]
This appears to be an incompatibility issue with newer versions of typescript. I didn't look into that much; but with the current version of typescript declared in the package.json file in @namnv04's repo, npm pulls in version 3.2.2. If you hardcode the version to 3.0.3, the issue goes away.

**Comment by lukeautry** [comment]
I think this should be fixed in 2.3.8 - please confirm.

**Comment by briepace** [comment]
2.3.8 fixed the issue.
